src/components/ContextBar.js

```diff
@@ -20,54 +36,114 @@ const contextBarStyles = {
+          ]),
+          side: 'bottom'
+        }, [
+          h(Clickable, {
+            'aria-label': 'Menu',
+            style: contextBarStyles.contextBarButton,
+            hover: contextBarStyles.hover,
+            tooltipSide: 'left',
+            tooltip: 'Menu',
+            tooltipDelay: 100
+          }, [icon('ellipsis-v', { size: 24 })])
+        ]),
+        h(Clickable, {
+          style: { ...contextBarStyles.contextBarButton, flexDirection: 'column', justifyContent: 'center', padding: '.75rem' },
+          hover: contextBarStyles.hover,
           tooltipSide: 'left',
-          onClick: () => setSharingWorkspace(true)
-        }, [makeMenuIcon('share'), 'Share']),
-        h(MenuButton, { disabled: true }, [makeMenuIcon('export'), 'Publish', comingSoon]),
-        h(MenuButton, {
-          disabled: !isOwner,
-          tooltip: !isOwner && 'You must be an owner of this workspace or the underlying billing project',
+          onClick: () => setCloudEnvOpen(!isCloudEnvOpen),
+          ...{ tooltip: 'Environment Configuration', tooltipDelay: 100 },
+          'aria-label': 'Environment Configuration'
+        }, [
+          img({ src: cloudIcon, style: { display: 'flex', margin: 'auto', height: 26, width: 26 } }),
+          getEnvironmentStatusIcons()
+        ]),
+        h(Clickable, {
+          style: { ...contextBarStyles.contextBarButton, color: !isTerminalEnabled ? colors.dark(0.7) : contextBarStyles.contextBarButton.color },
+          hover: contextBarStyles.hover,
           tooltipSide: 'left',
-          onClick: () => setDeletingWorkspace(true)
-        }, [makeMenuIcon('trash'), 'Delete Workspace'])
-      ]),
-      side: 'bottom'
-    }, [
-      h(Clickable, {
-        'aria-label': 'Menu',
-        style: contextBarStyles.contextBarButton,
-        hover: contextBarStyles.hover,
-        tooltip: 'Menu',
-        tooltipSide: 'left',
-        tooltipDelay: 100
-      }, [icon('ellipsis-v', { size: 24 })])
-    ]),
-    h(Clickable, {
-      style: contextBarStyles.contextBarButton,
-      hover: contextBarStyles.hover,
-      // TODO: add click handler
-      tooltip: 'Compute Configuration',
-      tooltipDelay: 100,
-      tooltipSide: 'left',
-      'aria-label': 'Compute Configuration'
-    }, [icon('cloudBolt', { size: 24 })]),
-    h(Clickable, {
-      style: contextBarStyles.contextBarButton,
-      hover: contextBarStyles.hover,
-      // TODO: add click handler
-      tooltip: 'Terminal',
-      tooltipSide: 'left',
-      tooltipDelay: 100,
-      'aria-label': 'Terminal'
-    }, [icon('terminal', { size: 24 })])
+          disabled: !isTerminalEnabled,
+          href: terminalLaunchLink,
+          onClick: window.location.hash === terminalLaunchLink && currentRuntime?.status === 'Stopped' ? () => startCurrentRuntime() : undefined,
+          tooltip: !isTerminalEnabled ? 'Terminal disabled due to current environment status' : 'Terminal',
```

Comment:
It's not just _status_ that can cause terminal to be disabled but also the kind of app (`Jupyter` vs. `RStudio` vs. `Galaxy`). Therefore, I wonder if something like below would be more accurate:
```suggestion
          tooltip: !isTerminalEnabled ? 'Terminal can be launched only with Jupyter environments when in running status.' : 'Terminal',
```

Reply:
I'll add the note about Jupyter environments, but its not true it can only be launched when the runtime is running